src/overlays/actors/ovl_En_Dg/z_en_dg.c

```diff
@@ -92,114 +147,1098 @@ static DamageTable D_8098C33C = {
     /* Powder Keg     */ DMG_ENTRY(0, 0x0),
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8098C45C[] = {
+static ActorAnimationEntryS D_8098C35C[] = {
+    { &D_060021C8, 1.0f, 0, -1, 0, 0 },   { &D_060021C8, 1.0f, 0, -1, 0, -6 },  { &D_06001BD8, 1.0f, 0, -1, 0, 0 },
+    { &D_06000998, 1.0f, 0, -1, 0, -6 },  { &D_06001FB0, 1.0f, 0, -1, 2, -6 },  { &D_06001FB0, 1.0f, 0, -1, 4, -6 },
+    { &D_06001048, 1.0f, 0, -1, 2, -6 },  { &D_06001348, 1.0f, 0, -1, 0, -6 },  { &D_06001048, 1.0f, 0, 27, 2, -6 },
+    { &D_06001048, 1.0f, 28, -1, 2, -6 }, { &D_06001048, 1.0f, 54, 54, 2, -6 }, { &D_060021C8, -1.5f, -1, 0, 0, -6 },
+    { &D_06001560, 1.0f, 0, -1, 2, 0 },   { &D_06001A84, 1.2f, 0, -1, 2, 0 },   { &D_060017C0, 1.2f, 0, -1, 2, 0 },
+    { &D_060021C8, 0.5f, 0, -1, 0, 0 },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_STOP),
 };
 
-#endif
+void func_80989140(SkelAnime* skelAnime, ActorAnimationEntryS arg1[], s32 arg2) {
+    f32 frameCount;
+
+    arg1 += arg2;
+    if (arg1->frameCount < 0) {
+        frameCount = SkelAnime_GetFrameCount(&arg1->animationSeg->common);
+    } else {
+        frameCount = arg1->frameCount;
+    }
+
+    SkelAnime_ChangeAnim(skelAnime, arg1->animationSeg, arg1->playbackSpeed + (BREG(88) * 0.1f), arg1->frame,
+                         frameCount, arg1->mode, arg1->transitionRate);
+}
+
+void func_80989204(EnDg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->collider.dim.pos.x = this->actor.world.pos.x;
+    this->collider.dim.pos.y = this->actor.world.pos.y;
+    this->collider.dim.pos.z = this->actor.world.pos.z;
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    if ((player->transformation == PLAYER_FORM_DEKU) && (this->actionFunc == func_8098AC34)) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    } else {
+        Collider_ResetCylinderAT(globalCtx, &this->collider.base);
+    }
+
+    if ((this->unk_290 != 1) && !(this->unk_280 & 2)) {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    } else {
+        Collider_ResetCylinderOC(globalCtx, &this->collider.base);
+    }
+
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 26.0f, 10.0f, 0.0f, 5);
+}
+
+void func_8098933C(EnDg* this, Vec3f* arg1) {
+    f32 sp24;
+    f32 sp20;
+    f32 sp1C;
+
+    if (this->actor.floorPoly != NULL) {
+        sp24 = this->actor.floorPoly->normal.x * 0.00003051851f;
+        sp20 = this->actor.floorPoly->normal.y * 0.00003051851f;
+        sp1C = this->actor.floorPoly->normal.z * 0.00003051851f;
```

Comment:
```suggestion
        sp24 = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.x);
        sp20 = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.y);
        sp1C = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.z);
```

Reply:
I didn't knew this macro existed. I love it.